### DISCUSSION TITLE
Guardian: Integration or Implementation of IPFS storage for VP/VC

### DISCUSSION THREAD

**Discussion by voycey** [hiero-ledger/hiero-improvement-proposals#290] [Ideas]
The guardian currently stores all VP/VC information locally in MongoDB, this data can be modified directly and unless published there is no method of linking the local VP's with anything on the blockchain reliably and this data is not immutable.

Changing the value in the database causes that changed value to appear in the trust-chain which breaks the whole trustless concept, also currently the only proof that is stored in the mintToken memo is the database id of the VP.

I am putting this out for discussion to see if anyone else is already looking at building such an implementation, if so it might be better to collaborate than roll our own.

If not then we will take this on as a team at TYMLEZ and develop something simple for IPFS storage for the VPs to allow for full transparency of the data

**Comment by anvabr** [comment]
SImilarly to #287, @voycey, thank you for the review and suggestion above. The team behind Guardian will take it onboard. However this is not the right place to submit bug reports and enhancement/feature requests for Guardian. The best way is to create an issue in the guardian repo https://github.com/hashgraph/guardian. Just click on 'Issues' and 'New Issue' and write-up your notes and Guardian product team will process it accordingly.

**Comment by anvabr** [comment]
Recorded in: https://github.com/hashgraph/guardian/issues/218
